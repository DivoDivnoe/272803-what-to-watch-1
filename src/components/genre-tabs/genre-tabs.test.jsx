import React from 'react';
import {create} from 'react-test-renderer';
import GenreTabs from './genre-tabs.jsx';

const mock = {
  genre: `horror`,
  clickHandler: jest.fn(),
  genres: [`all`, `crime`, `thriller`],
};

describe(`GenreTabs component`, () => {
  it(`renders correctly`, () => {
    const {genre, clickHandler, genres} = mock;

    const tree = create(
        <GenreTabs genre={genre} clickHandler={clickHandler} genres={genres} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
