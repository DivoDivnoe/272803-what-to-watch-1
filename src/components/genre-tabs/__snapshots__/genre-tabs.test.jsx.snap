// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreTabs component renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Thrillers
    </a>
  </li>
</ul>
`;
